### ISSUE TITLE
Future plans for versioneer

### ISSUE THREAD

**Issue by djhoese** [pydata/xarray#2853]
I have projects that use versioneer after finding out that xarray and dask used it. However, it seems the project is no longer actively maintained with the last commit happening almost 2 years ago:

https://github.com/warner/python-versioneer

There are two issues I've found with versioneer:

1. Wheels do not have a valid version number. They get `0+unknown` unless an `sdist` is built first to populate the egg info. Building sdist is not something that is always done by CI environments like travis. This also happens if someone installs the package from github `pip install git+https://github.com/pydata/xarray.git`. At least these are my findings with my satpy library.
2. `versioneer.py` has some style issues:
```
versioneer.py:941:-13592: W605 invalid escape sequence '\s'
versioneer.py:941:-13408: W605 invalid escape sequence '\s'
versioneer.py:941:-13228: W605 invalid escape sequence '\s'
versioneer.py:941:-11196: W605 invalid escape sequence '\d'
versioneer.py:941:-8162: W605 invalid escape sequence '\d'
```
This was worked around in xarray by adding a `noqa` to the top of the module.

So my question is, do the xarray devs (or dask devs) see this as a problem? What are your plans for the future? Have you enjoyed using versioneer?

I guess I should probably CC @warner on this too in case they are still active on github.

**Comment by xhochy** [comment]
I'm really happy with https://pypi.org/project/setuptools-scm/ in other projects, especially as its usage is normally a single line in setup.py.

**Comment by max-sixty** [comment]
Ditto @xhochy re `setuptools-scm` (though my experience is internally only, with more narrow requirements than a public project)

Our `setup.py`s contain only:

```python
    install_requires=['setuptools_scm'],
    use_scm_version=dict(version_scheme='post-release'),
```

**Comment by keewis** [comment]
I believe this has been fixed by moving to `setuptools-scm` in #3714.
